### PR TITLE
docs: Add documentation for installing and running Determined AI on WSL

### PR DESCRIPTION
**Pull request by sirredbeard**
## Description

This PR add documentation for installing and running Determined AI on WSL using both the deb/rpm packages and Docker containers.

Sections of the deb/rpm and Docker install pages were adapted with WSL-specifics learned when drafting both [this blog post](https://boxofcables.dev/running-determined-ai-on-wsl-2/) and [this project](https://github.com/sirredbeard/determined-wsl).

## Test Plan

The test plan for these changes include:

Configuring a new Windows VM with WSL2 and following the steps for both procedures to verify they work as stated.

## Checklist

- [X] Changes have been manually QA'd
- [NA] User-facing API changes need the "User-facing API Change" label.
- [NA] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [NA] Licenses should be included for new code which was copied and/or modified from any external code.

## Docs Checklist

- [x] Active voice applied
- [x] Style guide applied

### CODE REVIEW

**Comment by sirredbeard** [comment]
Rebased to remove merge conflict from re-org in #6819

**Review by tara-det-ai — CHANGES_REQUESTED** [review]
I applied some edits for style. Please check my [changes](https://github.com/determined-ai/determined/pull/6826/commits/d6da3bf0cef8d2ebaa3aacd2fb1110b6b94e22aa) to make sure I did not inadvertently introduce an error. We also need to edit the following file:

setup-cluster/deploy-cluster/on-prem/overview.rst

by adding "wsl" to the toctree

**Review by sirredbeard — COMMENTED** [review]
LGTM

File: docs/setup-cluster/deploy-cluster/on-prem/wsl.rst [hunk 1]
```diff
@@ -303,28 +341,30 @@ The following command starts the PostgreSQL container, replace ``<DB password>``
        -e POSTGRES_PASSWORD=<DB password> \
        postgres:10
 
-
 .. _docker_desktop_get_wsl_ip:
 
 Obtain the WSL IP address
 =========================
```
Comment (tara-det-ai) [inline docs/setup-cluster/deploy-cluster/on-prem/wsl.rst:347]: i missed this one, the "a" in "address" should be capitalized

**Comment by sirredbeard** [comment]
We also need to edit the following file:
> 
> setup-cluster/deploy-cluster/on-prem/overview.rst
> 
> by adding "wsl" to the toctree

My commit does that.

**Comment by sirredbeard** [comment]
> edit the following file:
> 
> setup-cluster/deploy-cluster/on-prem/overview.rst
> 
> by adding "wsl" to the toctree

[Done](https://github.com/determined-ai/determined/pull/6826/commits/93e42367b981c10924be699a047997f3e79d8402)

**Pull request merged**